### ISSUE TITLE
[perf regression] Softmax perf regression after triton config changed

### ISSUE THREAD

**Issue by weishi-deng** [intel/intel-xpu-backend-for-triton#5367]
### Describe the issue

In some models, softmax drops 70% of performance after the Pytorch PR (https://github.com/pytorch/pytorch/pull/162447), which changes the number of warps in reduction and causes register spill.

reproduce:

[softmax-new.py](https://github.com/user-attachments/files/23049941/softmax-new.py)

### Environment details

pytorch: f05e23e1bc1439e19145e43e8ffca0051cda2f33
triton: pytorch_triton_xpu-3.5.0+git1b0418a9

**Comment by whitneywhtsang** [comment]
No performance differences observed with and without https://github.com/pytorch/pytorch/pull/166107.

**Comment by weishi-deng** [comment]
Temporary workaround by https://github.com/pytorch/pytorch/pull/163908.
